components/map.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Mapsearch from './mapsearch.jsx';
+
+class Map extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onSearch = this.onSearch.bind(this);
+  }
+
+  componentDidMount() {
+    const theMap = this.refs.map;
+    const mapOptions = {
```

Comment:
Where are you using this??
Can theMap and mapOptions be moved to the constructor